internal/autodetect/detector/detector.go

```diff
@@ -0,0 +1,8 @@
+package detector
+
+import "context"
+
+type Detector interface {
+	IsRelevant(path string) (float64, error)
```

Comment:
could we rename `IsRelevant` to `Score`, or if you'd like `RelevancyScore` ?

EDIT: `Relevance() (float64, error)` also seems good